HopsanGUI/Utilities/HighlightingUtilities.cpp

```diff
@@ -348,60 +354,74 @@ CppHighlighter::CppHighlighter(QTextDocument *parent)
                     << "\\bslots\\b" << "\\bstatic\\b" << "\\bstruct\\b"
                     << "\\btemplate\\b" << "\\btypedef\\b" << "\\btypename\\b"
                     << "\\bunion\\b" << "\\bunsigned\\b" << "\\bvirtual\\b"
-                    << "\\bvoid\\b" << "\\bvolatile\\b" << "\\busing\\b";
-    foreach (const QString &pattern, keywordPatterns) {
+                    << "\\bvoid\\b" << "\\bvolatile\\b" << "\\busing\\b"
+                    << "\\bbool\\b" << "\\bif\\b" << "\\belse\\b"
+                    << "\\bfor\\b" << "\\bforeach\\b" << "\\bwhile\\b"
+                    << "\\bswitch\\b" << "\\bcase\\b" << "\\bdefault\\b"
+                    << "\\bbreak\\b" << "\\breturn\\b" << "\\bif\\()";
+    foreach (const QString &pattern, keywordPatterns)
+    {
         rule.pattern = QRegExp(pattern);
-        rule.format = keywordFormat;
-        highlightingRules.append(rule);
+        rule.format = mKeywordFormat;
+        mHighlightingRules.append(rule);
+    }
+
+    mHopsanKeywordFormat.setForeground(Qt::darkMagenta);
+    mHopsanKeywordFormat.setFontWeight(QFont::Normal);
+    QStringList hopsanKeywordPatterns;
+    hopsanKeywordPatterns << "\\bPort\\b" << "\\bFirstOrderTransferFunctionVariable\\b" << "\\bSecondOrderTransferFunctionVariable\\b"
+                          << "\\bFirstOrderTransferFunction\\b" << "\\bSecondOrderTransferFunction\\b"
+                          << "\\bFirstOrderFiler\\b" << "\\bSecondOrderFilter\\b";
+    foreach (const QString &pattern, hopsanKeywordPatterns)
+    {
+        rule.pattern = QRegExp(pattern);
+        rule.format = mHopsanKeywordFormat;
+        mHighlightingRules.append(rule);
     }
 
-    preProcessorFormat.setForeground(Qt::darkBlue);
-    preProcessorFormat.setFontWeight(QFont::Normal);
+    mPreProcessorFormat.setForeground(Qt::darkBlue);
+    mPreProcessorFormat.setFontWeight(QFont::Normal);
     keywordPatterns.clear();
     keywordPatterns << "^#?\\binclude\\b" << "^#?\\bifdef\\b" << "^#?\\bifndef\\b"
                     << "^#?\\belseif\\b" << "^#?\\belse\\b" << "^#?\\bendif\\b";
-    foreach (const QString &pattern, keywordPatterns) {
```

Comment:
One could also write 
if defined SOMETHING
if defined(SOMETHING)
same with else
Could be multiple whitespaces in there aswell so regexp maching is needed

Reply:
True, but I will ignore this for now. This only affects the highlighting anyway, so the code will work regardless.